src/master.rs

```diff
@@ -273,6 +273,43 @@ impl Master {
         })
     }
 
+    pub fn get_sdo(&mut self, slave_index: u16, sdo_position: u16) -> Result<SdoInfo> {
+        let mut sdo = ec::ec_ioctl_slave_sdo_t::default();
+        sdo.slave_position = slave_index;
+        sdo.sdo_position = sdo_position;
+        ioctl!(self, ec::ioctl::SLAVE_SDO, &mut sdo)?;
+        Ok(SdoInfo {
+            slave_position: sdo.slave_position,
+            position: sdo.sdo_position,
+            index: sdo.sdo_index,
+            max_subindex: sdo.max_subindex,
+            object_code: sdo.object_code,
+            name: raw_array_to_string(&sdo.name),
+        })
+    }
+
+    pub fn get_sdo_entry(
+        &mut self,
+        slave_index: u16,
+        sdo_spec: i32,
```

Comment:
the format for `sdo_spec` seems to be nontrivial: if I read correctly, negative numbers are "by position", positive are sdo indices. That should be wrapped in an enum on the Rust side.

Maybe even include the subindex?

Reply:
:+1:

Reply:
I added the enum `AdoEntryAddr`. What do you think?